pymeasure/display/widgets/log_widget.py

```diff
@@ -25,38 +25,148 @@
+    level_colors = {
+        "DEBUG": "DarkGray",
+        # "INFO": "Black",
+        "WARNING": "DarkOrange",
+        "ERROR": "Red",
+        "CRITICAL": "DarkRed",
+    }
+
+    html_replacements = {
+        "\r\n": "<br>  ",
+        "\n": "<br>  ",
+        "\r": "<br>  ",
+        "  ": "&nbsp;" * 2,
+        "\t": "&nbsp;" * 4,
+    }
+
+    def format(self, record):
+        formatted = super().format(record)
+
+        # Apply color if a level-color is defined
+        if record.levelname in self.level_colors:
+            formatted = f"<font color=\"{self.level_colors[record.levelname]}\">{formatted}</font>"
+
+        # ensure newlines and indents are preserved
+        for replacement in self.html_replacements.items():
+            formatted = formatted.replace(*replacement)
+
+        # Prepend the level as HTML comment
+        formatted = f"<!--{record.levelname}-->{formatted}"
+
+        return formatted
+
+
 class LogWidget(TabWidget, QtWidgets.QWidget):
     """ Widget to display logging information in GUI
 
     It is recommended to include this widget in all subclasses of
     :class:`ManagedWindowBase<pymeasure.display.windows.managed_window.ManagedWindowBase>`
     """
 
-    def __init__(self, name, parent=None):
+    fmt = '%(asctime)s : %(message)s (%(levelname)s)'
+    datefmt = '%m/%d/%Y %I:%M:%S %p'
+
+    tab_widget = None
+    tab_index = None
+
+    blink_qtimer = QtCore.QTimer()
+    blink_color = None
+    blink_state = False
+
+    def __init__(self, name, parent=None, fmt=None, datefmt=None):
+        if fmt is not None:
+            self.fmt = fmt
+        if datefmt is not None:
+            self.datefmt = datefmt
+
         super().__init__(name, parent)
         self._setup_ui()
         self._layout()
 
+        # Setup blinking
+        self.blink_qtimer.timeout.connect(self.blink)
+        self.handler.connect(self.blinking_start)
+
     def _setup_ui(self):
         self.view = QtWidgets.QPlainTextEdit()
         self.view.setReadOnly(True)
         self.handler = LogHandler()
-        self.handler.setFormatter(logging.Formatter(
-            fmt='%(asctime)s : %(message)s (%(levelname)s)',
-            datefmt='%m/%d/%Y %I:%M:%S %p'
+        self.handler.setFormatter(HTMLFormatter(
+            fmt=self.fmt,
+            datefmt=self.datefmt,
         ))
-        self.handler.connect(self.view.appendPlainText)
+        self.handler.connect(self.view.appendHtml)
 
     def _layout(self):
         vbox = QtWidgets.QVBoxLayout(self)
         vbox.setSpacing(0)
 
         vbox.addWidget(self.view)
         self.setLayout(vbox)
+
+    def blink(self):
+        if self.blink_state:
+            self.tab_widget.tabBar().setTabTextColor(
+                self.tab_index,
+                QtGui.QColor("black")
+            )
+        else:
+            self.tab_widget.tabBar().setTabTextColor(
+                self.tab_index,
+                QtGui.QColor(self.blink_color)
+            )
```

Comment:
```suggestion
        self.tab_widget.tabBar().setTabTextColor(
            self.tab_index,
            QtGui.QColor("black" if self.blinking_state else self.blink_color)
        )
```
Minor code optimization, if you like it.

Reply:
Yes, that's bit more compact, thanks